### DISCUSSION TITLE
About copy&paste

### DISCUSSION THREAD

**Discussion by siyou** [toeverything/blocksuite#8710] [Q&A]
When I paste an image from an image block, a new source ID is generated. Is this as expected?

**Comment by xiaogonggong-w** [comment]
> When I paste an image from an image block, a new source ID is generated. Is this as expected?当我从图像块粘贴图像时，会生成一个新的源 ID。这符合预期吗？

Hello, when pasting an image, I didn't upload a BlobSource. How did you capture the pasted image? If you could answer me, I would greatly appreciate it.

**Comment by raintoway** [comment]
> When I paste an image from an image block, a new source ID is generated. Is this as expected?

not , because the binary data of these two images is identical, when you create a new image from other image-block, it will compare with the blobSource in the global `blobSourceStore`. If there's a match, it will return the same sourceId.
